app/controllers/posts_controller.rb

```diff
@@ -65,6 +63,9 @@ def destroy
       format.html { redirect_to posts_url, notice: 'Post was successfully destroyed.' }
       format.json { head :no_content }
     end
+    @notifications=Notification.where(target_type:"posts").where(target_id:@post[:id])
+
+    @notifications.destroy_all
```

Comment:
Why?

Reply:
if someone commented on the post and after that user deletes that post then all the notifications related to that post should be deleted because these notifications point to nothing.

Reply:
If that is the case. Make a method in Notification Model move this code there. And then simply call that method here.

Reply:
Done